MekHQ/src/mekhq/gui/view/PersonViewPanel.java

```diff
@@ -227,9 +297,100 @@ private void initComponents() {
         gridBagConstraints.fill = GridBagConstraints.BOTH;
         gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
         add(txtFiller, gridBagConstraints);
+    }
+
+    private void setRibbons() {
+    	int maxRibbonsPerRow = 4;
 
+    	ArrayList<Award> awards = person.getAwards().stream().filter(a -> a.getRibbonFileName() != null)
```

Comment:
There's no need for a specific implementation of the List interface here. I would declare the variable as `List<Award>` and use the `Collectors.toList()` method.

Reply:
By changing this to
```java
List<Award> awards = person.getAwards().stream().filter(a -> a.getMedalFileName() != null).sorted().collect(Collectors.toList());
```

The following error appears:

Incompatible types. Required List<Award> but 'collect' was inferred to R: no instace(s) of type variable(s) T exist so that List<T> conforms to List.

Reply:
This is because Collectors::toList returns a value of type java.util.List which it's trying to assign to a variable that is of type java.awt.List because of the wildcard import of the java.awt package. You could either use the fully qualified name of the class (java.util.List<Award> awards = ....), remove the wildcard import and import the eight classes individually, or go back to what you had before. Developing the habit of using the interface that it makes the code more flexible, but in this case it's used entirely in a local context and there's no particular advantage to the abstraction.